pkg/internal/shim/shim.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shim
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/apparentlymart/go-versions/versions"
+	"github.com/hashicorp/hcl/v2"
+	tfregaddr "github.com/hashicorp/terraform-registry-address"
+	"github.com/hashicorp/terraform-svchost/disco"
+	tfaddr "github.com/opentofu/registry-address"
+	"github.com/pulumi/terraform/pkg/configs"
+	"github.com/pulumi/terraform/pkg/getproviders"
+)
+
+// A simple shim to access internals of opentofu for resolving versions.
```

Comment:
Any reason we've gone for `shim` here and not something like `opentofu`? Just feels like it'll be harder to see what's going on at a glance.

Reply:
Yeh shim feels consistent here. Bit sad we're now looking at opentofu and TF registries though. See the handling for ModuleSourceRegistry in tf.go.